src/hooks/collectible-info/use-collectible-info.hook.ts

```diff
@@ -0,0 +1,54 @@
+import { isNonEmptyArray } from '@apollo/client/utilities';
+import { useEffect, useState } from 'react';
+import { EMPTY, of } from 'rxjs';
+import { map, tap, finalize, catchError, switchMap } from 'rxjs/operators';
+
+import { fetchCollectibleInfo$ } from 'src/apis/objkt/index';
+
+import { CollectibleInfo } from '../../interfaces/collectible-info.interface';
+import { showErrorToast } from '../../toast/toast.utils';
+import { getAttributesInfo$, getAttributesWithRarity } from './collectible-info.utils';
+import { collectibleInfoInitialState } from './constants';
+
+export const useCollectibleInfo = (address: string, id: string) => {
+  const [collectibleInfo, setCollectibleInfo] = useState<CollectibleInfo>(collectibleInfoInitialState);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const subscription = fetchCollectibleInfo$(address, id)
+      .pipe(
+        tap(() => setIsLoading(true)),
+        switchMap(collectibleInfo => {
+          if (!isNonEmptyArray(collectibleInfo.attributes)) {
+            return of(collectibleInfo);
+          }
+
+          const attributeIds = collectibleInfo.attributes.map(({ attribute }) => attribute.id);
+
+          return getAttributesInfo$(attributeIds, isNonEmptyArray(collectibleInfo.galleries)).pipe(
+            map(attributesInfo => {
+              return {
+                ...collectibleInfo,
+                attributes: getAttributesWithRarity(attributesInfo, collectibleInfo)
+              };
+            })
+          );
+        }),
+        catchError(err => {
+          showErrorToast({ description: err.message });
+
+          return EMPTY;
+        }),
+        finalize(() => setIsLoading(false))
+      )
+      .subscribe(result => {
+        setCollectibleInfo(result);
+      });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
```

Comment:
Maybe we can move this logic to epic, dispatch an action and save result to `redux store` not to `react state`?

Reply:
Why should we do this?